### PR TITLE
new(ci): added github pages around rules overview file.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area documentation

**What this PR does / why we need it**:

This PR adds a github page around rules overview file.
Old file `rules_inventory/rules_overview.md` is removed because it is now useless.
NOTE: no file needs to be pushed; the overview is generated in the CI and deployed to gh pages without pushing anyhing on the repo.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+site_name: Falcosecurity Rules
+site_url: https://github.com/falcosecurity/rules
+nav:
+    - Home: index.md
```
Comment (FedeDP) [inline mkdocs.yml:4]: We have a single file, the overview one.

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+site_name: Falcosecurity Rules
```
Comment (FedeDP) [inline mkdocs.yml:1]: Mkdocs config file; we only have a single page, the overview one.

File: .github/workflows/pages.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+      
+      - name: Generate inventory
+        run: |
+          python rules_inventory/scripts/rules_overview_generator.py --rules_file=rules/falco_rules.yaml > docs/overview.md
```
Comment (FedeDP) [inline .github/workflows/pages.yaml:30]: We generate the overview page dynamically.

Comment (incertum) [reply .github/workflows/pages.yaml:30]: In a follow up PR we could find a new home for the python script? For example directly more hidden in the `.github`? Let me know and I was gonna update the script now that most rules are already tagged.

**Review by incertum — APPROVED** [review]
Thanks a bunch @FedeDP!

/approve

File: .github/workflows/pages.yaml (hunk 3 above)
Comment (FedeDP) [reply .github/workflows/pages.yaml:30]: We can do that for sure!

**Pull request merged**